pkg/api/validation/validation.go

```diff
@@ -372,6 +372,11 @@ func ValidateObjectMetaUpdate(newMeta, oldMeta *api.ObjectMeta, fldPath *field.P
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceVersion"), newMeta.ResourceVersion, "must be specified for an update"))
 	}
 
+	// Generation should be the same or at most incremented by one.
+	if newMeta.Generation != oldMeta.Generation && newMeta.Generation != oldMeta.Generation+1 {
```

Comment:
@bprashanth will know better but I would say this is the latter. Or at least this was always my understanding of generation. observedGeneration otoh should be allowed to increment more than one since that's what a controller has observed.